frontend/src/types/errorResponse.ts

```diff
@@ -0,0 +1,95 @@
+/**
+ * An error response returned by the API. This implements the error responses
+ * ADR, please refer to doc/adr/0012-error-responses.md for more information.
+ */
+export type ErrorResponse<
+  T extends keyof ErrorResponseDetail = "__fallback__",
+  Children = never,
```

Comment:
Currently not used, we will need to update this once we actually support/need nested errors. Not necessary in the first iteration IMO but wanted to already add some kind of marker.